src/calculator.py

```diff
@@ -116,91 +114,91 @@ def main():
+
+            investor_q = sess.query(Investor).filter(Investor.name == investment.name)
+            investor = investor_q.one()
+
+            logging.info(f"New mature investment: {investment.comment}")
+            logging.info(f" -- by {investor.name}")
+
+            if investment.response != "0":
+                response = reddit.comment(id=investment.response)
+            else:
+                response = EmptyResponse()
+
+            # If comment is deleted, skip it
+            try:
+                reddit.comment(id=investment.comment)
+            except:
+                logging.info(f" -- skipped (deleted comment)")
+                response.edit_wrap(message.deleted_comment_org)
+                continue
+
+            post = reddit.submission(investment.post)
+            upvotes_now = post.ups # <--- triggers a Reddit API call
+
+            # Updating the investor's balance
+            factor = calculate(upvotes_now, investment.upvotes)
+            amount = investment.amount
+            balance = investor.balance
+
+            new_balance = int(balance + (amount * factor))
+            change = new_balance - balance
+
+            # Updating the investor's variables
+            update = {
+                Investor.completed: investor.completed + 1,
+                Investor.balance: new_balance,
+            }
+            investor_q.update(update, synchronize_session=False)
+
+            # Editing the comment as a confirmation
+            text = response.body # <--- triggers a Reddit API call
+            if change > amount:
+                logging.info(f" -- gained {change}")
+                response.edit_wrap(message.modify_invest_return(text, change, new_balance))
+            elif change == amount:
+                logging.info(f" -- broke even ({change})")
+                response.edit_wrap(message.modify_invest_break_even(text, change, new_balance))
+            else:
+                lost_memes = int( amount - change )
+                logging.info(f" -- lost {lost_memes}")
+                response.edit_wrap(message.modify_invest_lose(text, lost_memes, new_balance))
+
+            sess.query(Investment).\
+                filter(Investment.id == investment.id).\
+                update({
+                    Investment.success: change > amount,
+                    Investment.done: True
+                }, synchronize_session=False)
+
+            sess.commit()
+
+            # Measure how long processing took
+            duration = stopwatch.measure()
+            logging.info(f" -- processed in {duration:5.2f}s")
+
+            # Report the Reddit API call stats
+            rem = int(reddit.auth.limits['remaining'])
+            res = int(reddit.auth.limits['reset_timestamp'] - time.time())
+            logging.info(f" -- API calls remaining: {rem:3d}, resetting in {res:3d}s")
         except Exception as e:
             logging.error(e)
             time.sleep(10)
+        finally:
```

Comment:
Woah, took me a while to find the initial try. Is it a good idea to have `try` catch an exception from ~90 lines. Maybe we can `try-except` some strange portions or functions, but not the whole calculate section? Nothing serious, just was thinking.

Reply:
Good suggestion, I agree that this try/catch block has probably grown a bit too long. I think it's out of the scope of this PR to fix it, though. But in a future PR we should probably make this loop look a bit more like `main.py`, with a relatively short outer loop (where the try/catch would be) that calls into a function like `process_investment` where the longer logic would be.

Reply:
I mean, we can be sure in math, we don't need to catch an exception from `2+2` operations. Heh